### PR TITLE
[Data][Docs] Standardize "Sorting, Shuffling, Repartitioning" references

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

This PR standardizes API references listed under "Sorting, Shuffling, Repartitioning".

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by angelinalg — APPROVED** [review]
I had just some nits.

File: python/ray/data/dataset.py [hunk 1]
```diff
@@ -907,28 +907,31 @@ def filter(
         return Dataset(plan, self._epoch, self._lazy, logical_plan)
 
     def repartition(self, num_blocks: int, *, shuffle: bool = False) -> "Dataset":
-        """Repartition the dataset into exactly this number of blocks.
+        """Repartition the :class:`Dataset` into exactly this number of :ref:`blocks <dataset_concept>`.
 
-        After repartitioning, all blocks in the returned dataset will have
-        approximately the same number of rows.
+        This method can be useful to tune the performance of your pipeline.
```
Comment (amogkam) [inline python/ray/data/dataset.py:912]: link to performance tuning?

File: python/ray/data/dataset.py [hunk 2]
```diff
@@ -942,7 +945,8 @@ def repartition(self, num_blocks: int, *, shuffle: bool = False) -> "Dataset":
                 minimizing data movement.
 
         Returns:
-            The repartitioned dataset.
+            The repartitioned :class:`Dataset`. After repartitioning, all blocks have
+            approximately the same number of rows.
```
Comment (amogkam) [inline python/ray/data/dataset.py:949]: is the all blocks have approximately the same number of rows true? I don't think it is right?

File: python/ray/data/dataset.py [hunk 3]
```diff
@@ -964,35 +968,31 @@ def random_shuffle(
         num_blocks: Optional[int] = None,
         **ray_remote_args,
     ) -> "Dataset":
-        """Randomly shuffle the elements of this dataset.
+        """Randomly shuffle the rows of this :class:`Dataset`.
 
         .. tip::
 
-            ``random_shuffle`` can be slow. For better performance, try
+            This method can be slow. For better performance, try
```
Comment (amogkam) [inline python/ray/data/dataset.py:975]: link to shuffling tuning guide?

File: python/ray/data/dataset.py [hunk 4]
```diff
@@ -2015,26 +2019,23 @@ def std(
...
+            >>> ds.sort("id", descending=True).take(3)
+            [{'id': 99}, {'id': 98}, {'id': 97}]
 
         Time complexity: O(dataset size * log(dataset size / parallelism))
 
         Args:
-            key: The column to sort by. To sort by multiple columns, use a map function
-                to generate the sort column beforehand.
+            key: The column to sort by. To sort by multiple columns, call
+                :meth:`Dataset.map` and generate a sort key.
```
Comment (amogkam) [inline python/ray/data/dataset.py:2034]: to generate a new column to sort by

**Review by amogkam — APPROVED** [review]

File: python/ray/data/dataset.py (hunk 2 above)
Comment (bveeramani) [reply python/ray/data/dataset.py:949]: Not sure. That's what our old documentation said.

**Comment by bveeramani** [comment]
Code format failures are from https://github.com/ray-project/ray/pull/37566 and https://github.com/ray-project/ray/pull/36980. See https://github.com/ray-project/ray/pull/37755

**Pull request merged**